commit d23cee157380456febd58439372e90ea73674cbe (refs/changes/98/498/1)
Author: Alex Wilson <alex.wilson@joyent.com>
Date:   2016-09-19T10:02:44-07:00 (3 years, 1 month ago)
    
    PUBAPI-1323 smartdc-auth locked key exception falling back to "Private key not available" message

diff --git a/lib/keypair.js b/lib/keypair.js
index 380af1d..df4f8a6 100644
--- a/lib/keypair.js
+++ b/lib/keypair.js
@@ -113,7 +113,8 @@ SDCKeyPair.prototype.createSign = function (opts) {
 
 	if (this.skp_private === undefined) {
 		throw (new Error('Private key for this key pair is ' +
-		    'unavailable'));
+		    'unavailable (because, e.g. only a public key was ' +
+		    'found and no matching private half)'));
 	}
 	var key = this.skp_private;
 	var keyId = this.getKeyId();
@@ -206,6 +207,26 @@ function SDCLockedKeyPair(kr, opts) {
 }
 mod_util.inherits(SDCLockedKeyPair, SDCKeyPair);
 
+SDCLockedKeyPair.prototype.createSign = function (opts) {
+	if (this.lkp_locked) {
+		throw (new Error('SSH private key ' +
+		    this.getPublicKey().comment +
+		    ' is locked (encrypted/password-protected). It must be ' +
+		    'unlocked before use.'));
+	}
+	return (SDCKeyPair.prototype.createSign.call(this, opts));
+};
+
+SDCLockedKeyPair.prototype.createRequestSigner = function (opts) {
+	if (this.lkp_locked) {
+		throw (new Error('SSH private key ' +
+		    this.getPublicKey().comment +
+		    ' is locked (encrypted/password-protected). It must be ' +
+		    'unlocked before use.'));
+	}
+	return (SDCKeyPair.prototype.createRequestSigner.call(this, opts));
+};
+
 SDCLockedKeyPair.prototype.canSign = function () {
 	return (true);
 };
diff --git a/package.json b/package.json
index 2956924..e37ff7b 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
     "name": "smartdc-auth",
     "description": "Joyent Authentication Library",
-    "version": "2.5.1",
+    "version": "2.5.2",
     "author": "Joyent (joyent.com)",
     "contributors": [
         "Mark Cavage",
